dev/breeze/src/airflow_breeze/breeze.py

```diff
@@ -302,7 +303,8 @@ def build_ci_image(
     build_cache: Optional[str],
     platform: Optional[str],
     debian_version: Optional[str],
-    upgrade_to_newer_dependencies: Optional[str],
+    upgrade_to_newer_dependencies: bool,
```

Comment:
```suggestion
    upgrade_to_newer_dependencies: bool,
```
This should be Optional[str] actually :)

Reply:
That was the fix I made here -> https://github.com/apache/airflow/pull/22597

Reply:
@potiuk Is it Optional[str] for prod image too?

Reply:
Yep

Reply:
I added more fixes on the parameter in #22649 :). You will need to rebase and resolve conflicts :)